src/node_kernels/MomentumBodyForceBoxNodeKernel.C

```diff
@@ -10,31 +10,103 @@
 #include "node_kernels/MomentumBodyForceBoxNodeKernel.h"
 #include "Realm.h"
 #include "SolutionOptions.h"
+#include "LowMachEquationSystem.h"
+#include "NaluEnv.h"
+#include "master_element/MasterElementFactory.h"
+#include "TimeIntegrator.h"
 
 #include "stk_mesh/base/MetaData.hpp"
 #include "stk_mesh/base/Types.hpp"
+#include "stk_mesh/base/Selector.hpp"
 #include "utils/StkHelpers.h"
+#include <stk_util/parallel/ParallelReduce.hpp>
+
+#include "ngp_utils/NgpLoopUtils.h"
+
+#include <fstream>
+#include <iomanip>
 
 namespace sierra {
 namespace nalu {
 
 MomentumBodyForceBoxNodeKernel::MomentumBodyForceBoxNodeKernel(
-  const stk::mesh::BulkData& bulk,
-  const std::string coordName,
+  Realm& realm,
```

Comment:
I would prefer if we didn't pass `Realm` into this object, but rather added a flags or a struct. In general I think having Realm be a god object that contains all the options is one of the things that makes nalu so confusing to work with, hard to read and very difficult to test. Any new code we add should steer away from that practice IMO.

Reply:
I also agreed with your sentiment, Phil. Realm does really complicate things... However it does look like fixing this would be a larger scope of work that this PR...

Reply:
I think there is general agreement that Realm and SolutionOptions are not great. You just have to look at the amount of scaffolding we need to create a unit test. However I am not convinced that having skinny structs is any better. You are just moving the problem from the constructor to another place where the struct gets populated. You just need to look at Nalu Parsing to see why that idea can quickly get out of hand. You could just pass the appropriate YAML node as an argument and let the class manage the data and associated checks it needs to do. Right now you've to change things at multiple places to add a parameter to a kernel. Structs don't eliminate the problem and at least I can't see the need for another object when the class can handle this responsibility. 

Of course the argument for not passing YAML nodes but use a struct is: what if we ever wanted to support some other format or use another library.

Reply:
In fact, passing the appropriate YAML node can be done quite easily with the current codebase quite easily. But is it ok to expose implementation details in Momentum EQS

Reply:
I was actually thinking about starting a github discussion about this very thing.  I agree that just passing yaml would be the best way to do things. Use Realm to store the algs and use factories to create them that take in yaml nodes.  I would like to detangle some of this incrementally throughout the year. I think the advantage to skinny structs is it is an incremental way of moving in that direction.  It's definitely not the solution, but I see it is a start toward encapsulation. If things don't depend on Realm but rather structs means that Realm can be changed easier without cascading consequences because the items the dependencies go through a filtering state as they leave Realm by constructing these skinny structs.  I'm sure there are several ways to go about it, but in general I feel that perpetuating the use of Realm as a mega container will only make the problem harder to solve in the long run.  I would love to see solution options completely removed. One can dream... 

@marchdf sorry to hijack your PR.  feel free to merge whenever you'd like.

Reply:
@psakievich Please let me know when you move this discussion to a separate Issue. Sorry from my end too @marchdf